As junior I want to edit the video details/information so i can sell better

javierLiarte opened the pull request:
https://www.pivotaltracker.com/story/show/155949343

javierLiarte committed:
Create new video edit detail page. First layout using angularjs-material for layout and UI elements

- Add angular material and material icons stylesheets to index.html and dist/index.html
- Fix DOCTYPE tag in index.html and dist/index.html
- Add video-detail-edit.controller to index.html
- Add Button variables to m4n and vimojo flavour sass variables file
- Update gulpfile; add gmapsApiKey to make-config task; add ng-map to join-dependencies sourceDeps; fix sourceSaas paths for gulp watch scss changes
- Add ngmap to package.json dependencies
- Update js/app.js config to setup angularjs-material theming
- Add videoDetailEdit state to app.js
- Update sass/style.scss: Fix components sass imports; Add video-detail-edit.view sass import
- Update video service with new getVideoLangs function to get video languages
- Create new video-detail-edit page using angularjs-material for layout and UI elements

javierLiarte committed:
Untrack css/style.css

javierLiarte committed:
Ignore css/style.css file

javierLiarte committed:
Merge branch 'dev' into 155949343_edit_video_detail_spike and improve edit detail page

- Merge last dev branch changes
- Create new loading-card directive using ng-material cards and material layout engine
- Add flex layout-fill to index.html main ui-view
- Refactor video-detail-edit view extracting partials templates
- Add UI improvements and fixes
- Refactor and clean video edit controller
- Create put function in api service
- Create update and getProductTypes functions in video service

javierLiarte committed:
Create new error404 directive and use in video detail and video detail edit pages. UI updates

- Fix loading-card view margins
- Add new directive error404 for showing 404 errors in platform
- Add default error404 title and message translations
- Add EditCtrl.editorRole flag to disable editing editor fields if user has not editor role
- Update reset icon for new video file
- Update video detail page to use new error404 directive and check for video existence

javierLiarte committed:
Update video edit comtroller for editor role checking

CodingCarlos committed:
Merge branch 'dev' into 'update-featured-videos'

javierLiarte committed:
Add video languages translations and video edit prevent form submission on validation errors

- Add video languages translations
- Add price fields initialization in video edit controller
- Add form and prevent form submission to edit video view
- UI updates

CodingCarlos committed:
Add new featured to video gallery

javierLiarte committed:
Add video edit translations. Add categories support

- Add video edit page translations
- Add hint tooltip to new video input
- Update edit video controller to support categories and check video maps before initialization

javierLiarte committed:
Add video edit upload new files handling

- Clean video service
- Update video service update to handle api error
- Remove unneeded ng-init clauses in featured and verified input elements
- Update VideoDetailEditController for handling file uploads; Add fixes for initial values
- Refactor and clean VideoDetailEditController
- Refactor api to reuse file upload code in update mehtod

javierLiarte committed:
Merge branch 'dev' into 155949343_edit_video_detail_spike

javierLiarte committed:
Fix video player and poster not updating on backend video update

javierLiarte committed:
Add map and place autocompletion handling for video details edit page

- Update index.html google maps script include; add trust-as-resource filter loading
- Add new filter trustAsResourceUrl to allow URI constructions in html
- Update session service to get current lang from $translate
- Update main controller to var with set session currentLang
- Clean video edit detail view
- Update video edit detail controller to handle map and places components
- Update video map partial to add places input and map marker
- Update basic details partial disabling location input

javierLiarte committed:
Merge branch '155949343_edit_video_detail_spike' into update-featured-videos

javierLiarte committed:
Merge pull request #46 from Videona/update-featured-videos

Update featured videos and gallery

javierLiarte committed:
Update video details edit page to get video categories from backend

- Add video service getVideoCategories method to get video categories from backend
- Update video detail edit controller initSelectMaps to get video categories from video service
- Update basic details partials video category select component
- Add translations for video categories

javierLiarte committed:
Add init for text fields

javierLiarte committed:
Update video details edit page to set max video upload size hint

- Add maxVideoUploadByteSize config var to gulpfile make-config task
- Update video file partial to use EditCtrl.maxVideoUploadSize
- Update video detail edit controller to set maxVideoUploadSize with config maxVideoUploadByteSize var; clean unused services and vars

javierLiarte committed:
Add localization support for material dates using moment

- Add moment dependency to package.json and gulpfile external modules
- Add mdDateLocaleProvider set up to app.js config phase

javierLiarte committed:
UI fixes

- Update app.js material theming set up to use blue palette for primary palette
- Fix input margins and paddings
- Update video-edit-page__aditional-details paddings
- Remove video thumbnails header texts
- Update upload file drop areas style
- Update hint color
- Fix mobile responsive step adding md-input-container to places-auto-complete input
- Add $ultraLightGray to vimojo and m4n variables

javierLiarte commented:
Still pending showing error/toast if form doesn't validate

CodingCarlos commented on flavour/vimojo/sass/base/_variables.scss:
```diff
@@ -15,6 +16,7 @@ $secondaryColorDisabled: #000000;
 /* Grey */
 $grey: #5A6978;
 $lightGrey: #B6B6B6;
+$ultraLightGray: #ededed;
```
Grey vs Gray, both correct, both used. Shall we unify it?

DevStarlight replied:
w/e but hex in mayus =)

CodingCarlos commented on services/api.js:
```diff
@@ -10,6 +10,7 @@
 			upload: upload,
 			get: get,
 			post: post,
+			put,
```
Unconsistent with the other proprties definition. Change to 
```
put: put
```
Or all the other porperties to follow the ES6 criteria.

javierLiarte replied:
done

CodingCarlos commented on services/video.js:
```diff
@@ -20,6 +24,51 @@ function videoService(api) {
+	}
+
+	function getVideoCategories() {
+		return new Promise((resolve, reject) => {
```
Be carefull about using native promises instead of [$q](https://docs.angularjs.org/api/ng/service/$q), since there are not watched by angular's $scope, and values inside them (and also changes in that values) might not be consistent with the code. In some cases you'll have to execute `$rootScope.apply()` or `$rootScope.digest()`, which in 99% of cases is a very bad idea (becouse of the $scope circular reference implementation).

$q is the angular's promises implementation, which is fully likned with $scope, and (if using promises) is the better approach for compatibility and avoid value duplicity and data losts.

DevStarlight replied:
ES6 as well...

javierLiarte replied:
updated!

CodingCarlos commented on services/video.js:
```diff
@@ -6,6 +6,10 @@ function videoService(api) {
+		update,
+		getVideoLangs,
+		getProductTypes,
+		getVideoCategories,
```
Unconsistent with the other proprties definition. Change the new props to ES5 syntax (as the other get or reset methods), or all the other porperties to follow the ES6 criteria, but not mixed.

javierLiarte replied:
updated!

CodingCarlos commented on pages/video-detail-edit/video-detail-edit.view.html:
```diff
@@ -0,0 +1,74 @@
+			</div>
+
+			<div flex="100">
+				<ng-include src="'pages/video-detail-edit/partials/video-map.html'"></ng-include>
```
Even when it works, ng-include is known for make the digest cycles slow, and for cause so much bugs in production for forgiven things inside templates, or for parent scope's unexpected changes. Instead of that, there are recomended to create directives (or components), and pass as a parameter whatever it need inside.

This thing is not much mor performant, but cleaner, since this new directives are totally individual, and don't depend of a parent $scope controller to exist, but it might be passed to its own scope, wich make it very easy to mantain. Also, as it creates a new scope, there's not danger for inner stuff to be changed from outside this template, and enable a way to link information which shall be shared (in this case, the controller instance, or its props/params)

[Some literature about this](http://www.codelord.net/2016/11/17/avoiding-ng-include-for-elegenace-and-performance/). Not a real thing to change now, but just to keep in mind ;)

javierLiarte replied:
ok, we'll investigate further. I've already had this in mind since several partials (as map) could be used in other places (such as video detail page). This is a fast and working first pass/approach that simplefies and better structure a complex view like this by now

CodingCarlos commented on pages/video-detail-edit/video-detail-edit.view.scss:
```diff
@@ -0,0 +1,189 @@
+    width: 100%;
+  }
+
+  // TODO: move upper!
```
Don't forget this comment ;)

javierLiarte replied:
Jesus told me he was to take care of this. @DevStarlight count on me when doing this and adapt other components or views

CodingCarlos reviewed (approved):
Wow. So much work here! Really proud to see how fast you've adapted to Angular's workflow! 

There are some comments, which are not very important. Some of them are just code beauty stuff, and other are details of angular's inner work, which are not a problem here, but might be in some moment ;)

javierLiarte committed:
Add video location data handling; Update video detail edit form validation to show an error if does not validate

- Renamed EditCtrl.video.location to EditCtrl.video.locationName
- Update video detail edit controller to set video location data for api video update call; modify update function to show error if form does not validate

javierLiarte committed:
Add instant translations for video details edit controller messages

javierLiarte committed:
Fix remove custom thumbnail icon class

javierLiarte committed:
Update video detail edit controller to center and zoom map if video has location defined

javierLiarte committed:
Add missing string translations

javierLiarte committed:
Update .md-chips style

DevStarlight commented on flavour/m4n/sass/base/_variables.scss:
```diff
@@ -15,6 +15,7 @@ $secondaryColorDisabled: #000000;
 /* Grey */
 $grey: #5A6978;
 $lightGrey: #E6E9EC;
+$ultraLightGray: #ededed;
```
hex in mayus plix plox!

DevStarlight commented on gulpfile.js:
```diff
@@ -53,6 +55,8 @@ gulp.task('flavour', function () {
 	'../node_modules/ng-file-upload/dist/ng-file-upload.min.js',
 	'../node_modules/ng-file-upload/dist/ng-file-upload-shim.min.js',
+	'../node_modules/ngmap/build/scripts/ng-map.min.js',
+	'../node_modules/moment/min/moment-with-locales.min.js',
```
There's already a script included in index.html with the href of the moment cdn file. Should we remove it from there since it's here?

CodingCarlos replied:
This is for production index.html (/dist) and it's thought for libraries without CDN. 

Maybe it's not a bad idea update the /dist/index.html to include the CDN files instead of adding this dependencies here 🤔

DevStarlight commented on js/app.js:
```diff
@@ -76,6 +81,64 @@
 			});
 	}
+
+	function setupMaterialTheming($mdThemingProvider) {
```
consider moving this function to another file

CodingCarlos replied:
We could even make it part of flavour files 🤖

javierLiarte replied:
yep, this is the first test of material theming. This indeed has to be handled for m4n flavour

DevStarlight commented on package.json:
```diff
@@ -46,7 +46,9 @@
     "buffer-to-vinyl": "^1.1.0",
     "gulp": "^3.9.1",
     "gulp-ng-config": "^1.5.0",
+    "moment": "^2.22.0",
```
why adding this library if we're loading that from a CDN?

javierLiarte replied:
are we? I didn't know

DevStarlight commented on pages/video-detail-edit/video-detail-edit.view.scss:
```diff
@@ -0,0 +1,201 @@
+  .new-thumbnail-input {
+    flex-grow: 1;
+    display: flex;
+    flex-direction: column;
```
`@include flex-direction-column` for full compatibility plix plox!

javierLiarte replied:
thx

DevStarlight committed:
Merge branch 'dev' of github.com:Videona/platform-frontend into 155949343_edit_video_detail_spike

# Conflicts:
#	components/video-list/video-list.controller.js
#	components/video-list/video-list.js
#	js/app.js

DevStarlight committed:
Setup v0.5.0 - Add video search functionality

javierLiarte committed:
Update spanish VIDEO_DETAIL_SUMMARY translation to Descripción

javierLiarte committed:
Merge branch '155949343_edit_video_detail_spike' of github.com:Videona/platform-frontend into 155949343_edit_video_detail_spike

javierLiarte committed:
Update with code review suggested issues

- Remove moment dependency from package and gulpfile as it's already loaded from CDN in index.html
- Update returned object definitions in video and api services
- Update video service to use $q for promises

javierLiarte commented:
Thx, not much different of the rest of the webs I developed before :p 

Included some suggestions from review.

I'll fix @alvaromart and ruth QA issues, update versioning and merge

javierLiarte committed:
Update toast to show fixed on top right

javierLiarte committed:
Update save spinner button style

javierLiarte committed:
Update video detail edit action buttons style; fix update vide detail page partials html

javierLiarte committed:
Update check edit access to not be strict on video owner comparison

javierLiarte committed:
Merge branch 'dev' into 155949343_edit_video_detail_spike

javierLiarte committed:
Fix app setup

javierLiarte committed:
Add missing ngMap deppendency to app.js

javierLiarte committed:
Fix Dockerfile missing build args
